### DISCUSSION TITLE
Is there a native way of making POST requests

### DISCUSSION THREAD

**Discussion by harblaith7** [nuxt/nuxt#16699] [Questions] [answered]
For GET requests, we can use `useFetch` or `useAsyncData`, but is there something for POST requests? Would we just use `axios` for this? If so, how do we add it to our app? I can't seem to get it to work

**Comment by wobsoriano** [comment, marked as answer]
Just use `$fetch`. It's a wrapper for `fetch` in the browser and `node-fetch` in server.

```html
<script setup lang="ts">
const todo = ref({
  title: 'A title',
  completed: false
})

const addTodo = async () => {
  const result = await $fetch('/api/todos', {
    method: 'POST',
    body: todo.value
  })
}
</script>
````

https://github.com/unjs/ohmyfetch

**Comment by RSS1102** [comment]
Hello, I use `useAsyncData ` to make a `post `request and carry parameters, but my server successfully responded to the request, but the parameters are `{}`. What's wrong with me?

nuxt:(server get` {} `)
``` ts
  const { data } = await useAsyncData<BlogsTree[]>("blogsContent", () =>
    $fetch(ENV_VITE_URL + "blogs/getBlogsContent", {
      method: "POST",
      body: { title: "2222222" },
    })
  );
```
nestjs:
![image](https://user-images.githubusercontent.com/81673017/193526030-fdc2c3a2-f337-4b3d-8a74-0ecae45a1999.png)

---
postman:(server get` {title: "1" }`)
and i use **postman** can get successful.
![image](https://user-images.githubusercontent.com/81673017/193526212-8fca187e-1d4d-475e-89f0-05f4c92aa1e2.png)

**Reply by RSS1102** [reply]
> 你好，我`useAsyncData `是用来发出`post `请求并带参数的，但是我的服务器成功响应了请求，但是参数是`{}`. 我怎么了？
> 
> nuxt:(服务器获取`{}`)
> 
> ```ts
>   const { data } = await useAsyncData<BlogsTree[]>("blogsContent", () =>
>     $fetch(ENV_VITE_URL + "blogs/getBlogsContent", {
>       method: "POST",
>       body: { title: "2222222" },
>     })
>   );
> ```
> 
> 巢穴： ![图片](https://user-images.githubusercontent.com/81673017/193526030-fdc2c3a2-f337-4b3d-8a74-0ecae45a1999.png)
> 
> 邮递员：（服务器获取` {title: "1" }`） 和我使用**邮递员**可以成功。 ![图片](https://user-images.githubusercontent.com/81673017/193526212-8fca187e-1d4d-475e-89f0-05f4c92aa1e2.png)

https://github.com/nuxt/framework/discussions/7966 i do it
